### PR TITLE
Add adaptiveStreamSetting for pausing video when tab in background

### PR DESCRIPTION
**Pull request by lukasIO**
Adds an additional setting to adaptive stream for pausing video when the browser tab is in the background/hidden.
Defaults to true.

`adaptiveStream: { pauseVideoInBackground: true}`

### CODE REVIEW

File: src/room/track/Track.ts [hunk 1]
```diff
@@ -179,8 +183,20 @@ export class Track extends (EventEmitter as new () => TypedEventEmitter<TrackEve
...
+  protected appVisibilityChangedListener = () => {
+    if (this.backgroundTimeout) {
+      clearTimeout(this.backgroundTimeout);
+    }
+    // delay app visibility update if it goes to hidden
+    // update immediately if it comes back to focus
+    if (document.visibilityState === 'hidden') {
+      this.backgroundTimeout = setTimeout(
+        () => this.handleAppVisibilityChanged(),
+        BACKGROUND_REACTION_DELAY,
```
Comment (davidzhao) [inline src/room/track/Track.ts:195]: is the purpose of the timeout so that if you are switching tabs, it doesn't pause too aggressively?

Comment (lukasIO) [reply src/room/track/Track.ts:195]: excactly. delay pausing when going into hidden state, but jump straight back if leaving hidden state

File: src/room/track/RemoteVideoTrack.ts [hunk 2]
```diff
@@ -188,8 +183,6 @@ export default class RemoteVideoTrack extends RemoteTrack {
   protected async handleAppVisibilityChanged() {
     await super.handleAppVisibilityChanged();
     if (!this.isAdaptiveStream) return;
-    // on desktop don't pause when tab is backgrounded
-    if (!isMobile()) return;
```
Comment (lukasIO) [inline src/room/track/RemoteVideoTrack.ts:192]: @davidzhao this most likely caused the safari visibility bug. Am not able to reproduce the issue on this branch.

**Comment by lukasIO** [comment]
instead of a boolean we could also let this setting have three different options like `mobile | always | never` in order to differentiate between desktop and mobile usage.

**Review by davidzhao — APPROVED** [review]
lg!

**Pull request merged**